addon/index.ts

```diff
@@ -22,17 +25,36 @@ window.addEventListener('storage', function ({ key, newValue }) {
     return;
   }
 
+  // skip if newValue is null
+  if (newValue === null) {
+    return;
+  }
+
   localStorageCache.set(key, jsonParseAndFreeze(newValue));
 });
 
-export default function localStorageDecoratorFactory(...args) {
+export default function localStorageDecoratorFactory(
+  ...args: DecoratorArgs
+): void;
+
+export default function localStorageDecoratorFactory(
+  customLocalStorageKey: string
+): (...args: DecoratorArgs) => void;
+
+export default function localStorageDecoratorFactory(
+  ...args: [string] | DecoratorArgs
+): PropertyDescriptor | ((...args: DecoratorArgs) => void) {
   const isDirectDecoratorInvocation = isElementDescriptor(...args);
   const customLocalStorageKey = isDirectDecoratorInvocation
     ? undefined
-    : args[0];
+    : (args[0] as string);
```

Comment:
Would be nice if we could use a type predicate here to discriminate between the two arg sets